### PR TITLE
gas miners only update overlays when they actually need to

### PR DESCRIPTION
**Pull request by silicons**
the same reason flightsuits were crashing downstreams is the reason gas miners are crashing upstream.
Wew!
closes #28112

### CODE REVIEW

File: code/modules/atmospherics/machinery/other/miner.dm [hunk 1]
```diff
@@ -40,31 +40,41 @@
...
-		broken = TRUE
+		set_broken(TRUE)
 		return FALSE
 	if(broken)
-		broken = FALSE
+		set_broken(FALSE)
 		broken_message = ""
 	return TRUE
 
+/obj/machinery/atmospherics/miner/proc/set_active(setting)
```
Comment (ChangelingRain) [inline code/modules/atmospherics/machinery/other/miner.dm]: unused?

Comment (kevinz000) [inline code/modules/atmospherics/machinery/other/miner.dm]: @ChangelingRain I forgot to make it set active to true on init woops

File: code/modules/atmospherics/machinery/other/miner.dm [hunk 2]
```diff
@@ -106,7 +120,6 @@
 
 /obj/machinery/atmospherics/miner/process()
 	update_power()
-	update_icon()
```
Comment (RemieRichards) [inline code/modules/atmospherics/machinery/other/miner.dm]: Why was this merged, ever.

Comment (lzimann) [inline code/modules/atmospherics/machinery/other/miner.dm]: Something something I'm stupid

Comment (Cyberboss) [inline code/modules/atmospherics/machinery/other/miner.dm]: REEEEE

Comment (kevinz000) [inline code/modules/atmospherics/machinery/other/miner.dm]: I'm sorry >_<

File: code/modules/atmospherics/machinery/other/miner.dm (hunk 1 above)
Comment (lzimann) [inline code/modules/atmospherics/machinery/other/miner.dm]: This proc is only used once. It's not needed.

**Pull request merged**